### PR TITLE
Fix broken subscriptions

### PR DESCRIPTION
**Pull request by longfin**
This PR fixes broken subscriptions using the following steps:

1. it unifies `IHostBuilder` in `Program.cs` to re-use `NineChroniclesNodeService` initialized by `.UseNineChroniclesNode()`.
2. it converts global context lookup from using `IResolveFieldContext.Source` to normal properties injected by DI.
    - To accomplish it, I made some changes on `GraphQLTestBase` too.
3. At last, it integrates `StandaloneSubscription.RegisterTipChangedSubscription` to its constructor since we can ignore the edge case about lazy initialization of `NineChroniclesNodeService` now.

### CODE REVIEW

File: NineChronicles.Headless.Tests/GraphTypes/GraphQLTestBase.cs [hunk 1]
```diff
@@ -71,46 +63,43 @@ public GraphQLTestBase(ITestOutputHelper output)
...
             var configuration = configurationBuilder.Build();
 
             var services = new ServiceCollection();
             services.AddSingleton(StandaloneContextFx);
             services.AddSingleton<IConfiguration>(configuration);
+            services.AddSingleton(_ => ServiceBuilder.CreateNineChroniclesNodeService(genesisBlock, new PrivateKey()));
             services.AddGraphTypes();
             services.AddLibplanetExplorer<NCAction>();
             services.AddSingleton<StateQuery>();
+            services.AddSingleton(_ => ncService);
```
Comment (moreal) [inline NineChronicles.Headless.Tests/GraphTypes/GraphQLTestBase.cs:86]: How about using `ServiceCollection.AddSingleton<TService>(IServiceCollection, TService)`? Because it doesn't use anything from the `_`, `provider` parameter.

File: NineChronicles.Headless.Tests/GraphTypes/StandaloneMutationTest.cs [hunk 2]
```diff
@@ -197,20 +180,14 @@ public async Task Transfer()
...
-                MakeGenesisBlock(
-                    default,
-                    goldCurrency,
-                    ImmutableHashSet<Address>.Empty
-                );
-            NineChroniclesNodeService service = ServiceBuilder.CreateNineChroniclesNodeService(genesis, new PrivateKey());
-            StandaloneContextFx.NineChroniclesNodeService = service;
-            StandaloneContextFx.BlockChain = service.BlockChain;
+            NineChroniclesNodeService service = StandaloneContextFx.NineChroniclesNodeService!;
+            BlockChain<PolymorphicAction<ActionBase>> blockChain = StandaloneContextFx.BlockChain!;
```
Comment (moreal) [inline NineChronicles.Headless.Tests/GraphTypes/StandaloneMutationTest.cs:184]: How about using `GraphQLTestBase.BlockChain`?

File: NineChronicles.Headless.Tests/GraphTypes/StandaloneMutationTest.cs [hunk 3]
```diff
@@ -132,27 +122,20 @@ public async Task ActivateAccount()
...
```
Comment (moreal) [inline NineChronicles.Headless.Tests/GraphTypes/StandaloneMutationTest.cs:133]: How about using `GraphQLTestBase.BlockChain`?

File: NineChronicles.Headless/GraphTypes/ActionMutation.cs [hunk 4]
```diff
@@ -86,8 +83,8 @@ public ActionMutation()
                             name = avatarName,
                         };
 
-                        var actions = new PolymorphicAction<ActionBase>[] { action };
-                        Transaction<PolymorphicAction<ActionBase>> tx = blockChain.MakeTransaction(privateKey, actions);
+                        var actions = new NineChroniclesActionType[] { action };
+                        Transaction<NineChroniclesActionType> tx = blockChain.MakeTransaction(privateKey, actions);
```
Comment (moreal) [inline NineChronicles.Headless/GraphTypes/ActionMutation.cs:87]: Maybe, we should unify the alias name, `NineChroniclesActionType` or `NCAction`

File: NineChronicles.Headless.Tests/GraphTypes/GraphQLTestBase.cs (hunk 1 above)
Comment (longfin) [reply NineChronicles.Headless.Tests/GraphTypes/GraphQLTestBase.cs:86]: `.AddSingleton<T>()` is generally proper, but we can't use that because `NineChroniclesNodeService` couldn't be constructed by DI. (only `NineChroniclesNodeService.Create()` allowed yet) we should need to more refactor for it. 😓

File: NineChronicles.Headless.Tests/GraphTypes/StandaloneMutationTest.cs (hunk 3 above)
Comment (longfin) [reply NineChronicles.Headless.Tests/GraphTypes/StandaloneMutationTest.cs]: Applied! https://github.com/planetarium/NineChronicles.Headless/pull/481/commits/fe83722a3b00f38768986bbf436d85adbca774dc

File: NineChronicles.Headless.Tests/GraphTypes/StandaloneMutationTest.cs (hunk 2 above)
Comment (longfin) [reply NineChronicles.Headless.Tests/GraphTypes/StandaloneMutationTest.cs]: Applied! https://github.com/planetarium/NineChronicles.Headless/pull/481/commits/fe83722a3b00f38768986bbf436d85adbca774dc

File: NineChronicles.Headless/GraphTypes/ActionMutation.cs (hunk 4 above)
Comment (longfin) [reply NineChronicles.Headless/GraphTypes/ActionMutation.cs]: Applied. https://github.com/planetarium/NineChronicles.Headless/pull/481/commits/2538e1de46714dd554bc8d723787675e01287e3e

File: NineChronicles.Headless.Tests/GraphTypes/GraphQLTestBase.cs (hunk 1 above)
Comment (moreal) [reply NineChronicles.Headless.Tests/GraphTypes/GraphQLTestBase.cs:86]: Oh, yes. As you said, `NineChroniclesNodeService` couldn't be constructed by DI. But what I wanted to suggest was

```suggestion
            services.AddSingleton(ncService);
```

not

```csharp
services.AddSingleton<NineChroniclesNodeService>();
```

How do you think about this?

Comment (longfin) [reply NineChronicles.Headless.Tests/GraphTypes/GraphQLTestBase.cs]: I see, fixed. https://github.com/planetarium/NineChronicles.Headless/pull/481/commits/7f11f097f211230490a54cef8427653fefd61e60

File: NineChronicles.Headless.Executable/Program.cs [hunk 5]
```diff
@@ -218,10 +218,33 @@ static void ConfigureSentryOptions(SentryOptions o)
...
             Log.Logger = loggerConf.CreateLogger();
 
-            var tasks = new List<Task>();
+            if (appProtocolVersionToken is null)
+            {
+                throw new CommandExitedException(
+                    "--app-protocol-version must be present.",
+                    -1
+                );
+            }
```
Comment (moreal) [inline NineChronicles.Headless.Executable/Program.cs:227]: In history, `appProtocolVersionToken` was nullable because *libplanet node* mode was optional and it was executed through HTTP API. But, the `appProtocolVersionToken` has been required since #420 merged. So how about change it into non-nullable type instead?

File: NineChronicles.Headless.Executable/Program.cs [hunk 6]
```diff
@@ -218,10 +218,33 @@ static void ConfigureSentryOptions(SentryOptions o)
...
+
+            if (genesisBlockPath is null)
+            {
+                throw new CommandExitedException(
+                    "--genesis-block-path must be present.",
+                    -1
+                );
+            }
```
Comment (moreal) [inline NineChronicles.Headless.Executable/Program.cs:235]: This comment has same intention with the comment about `appProtocolVersionToken`.

Comment (longfin) [reply NineChronicles.Headless.Executable/Program.cs]: Added. https://github.com/planetarium/NineChronicles.Headless/pull/481/commits/19b87259e7df2b0d4f2d676410c46d37d4fa60b4

File: NineChronicles.Headless.Executable/Program.cs (hunk 5 above)
Comment (longfin) [reply NineChronicles.Headless.Executable/Program.cs]: Added. https://github.com/planetarium/NineChronicles.Headless/pull/481/commits/19b87259e7df2b0d4f2d676410c46d37d4fa60b4

**Review by moreal — APPROVED** [review]

**Review by riemannulus — APPROVED** [review]

**Review by limebell — APPROVED** [review]
